File: sol/estruturas/1548 - Fila do Recreio.py
# 1548 - Fila do Recreio
# https://www.urionlinejudge.com.br/judge/pt/problems/view/1548


def notas_iguais(notas_a, notas_b):
    'Retorna o número de notas iguais na mesma posição.'
    return sum(1 for a, b in zip(notas_a, notas_b) if a == b)


def main():
    n = int(input())
    for _ in range(n):
        _ = input()
        notas = [int(x) for x in input().split(' ')]
        sorted_notas = sorted(notas, reverse=True)
        print(notas_iguais(notas, sorted_notas))

main()

File: sol/estruturas/1548 - Fila do Recreio.rb
# 1548 - Fila do Recreio
# https://www.urionlinejudge.com.br/judge/pt/problems/view/1548

def notas_iguais(notas_a, notas_b)
    'Retorna o número de notas iguais com o mesmo índice.'
    notas_a.zip(notas_b).count { |a, b| a == b }
end

def notas_iguais2(notas_a, notas_b)
    'Retorna o número de notas iguais com o mesmo índice.
    Essa versão é mais rápida, mas eu não gostei muito.'
    n = notas_a
      .zip(notas_b)
      .map { |a, b| a == b ? 1 : 0 }
      .reduce(:+)
end

def main
    n = gets.to_i
    n.times do
        _ = gets.to_i
        notas = gets.split(' ').map(&:to_i)
        puts(notas_iguais(notas, notas.sort.reverse))
    end
end
  
main
